clients/imodels-client-management/src/operations/changeset-group/ChangesetGroupOperations.ts

```diff
@@ -0,0 +1,87 @@
+ *--------------------------------------------------------------------------------------------*/
+import { ChangesetGroupResponse, ChangesetGroupsResponse, EntityListIteratorImpl, OperationsBase } from "../../base/internal";
+import { AuthorizationCallback, EntityListIterator, HeaderFactories } from "../../base/types";
+import { ChangesetGroup } from "../../base/types/apiEntities/ChangesetGroupInterfaces";
+import { IModelsClient } from "../../IModelsClient";
+import { OperationOptions } from "../OperationOptions";
+import { getUser } from "../SharedFunctions";
+
+import { GetChangesetGroupListParams, GetSingleChangesetGroupParams } from "./ChangesetGroupOperationParams";
+
+export class ChangesetGroupOperations<TOptions extends OperationOptions> extends OperationsBase<TOptions> {
+  constructor(
+    options: TOptions,
+    private _iModelsClient: IModelsClient
+  ) {
+    super(options);
+  }
+
+  /**
+   * Gets Changeset Groups for a specific iModel. Wraps the
+   * {@link https://developer.bentley.com/apis/imodels-v2/operations/get-imodel-changeset-groups/ Get iModel Changeset Groups}
+   * operation from iModels API.
+   * @param {GetChangesetGroupListParams} params parameters for this operation. See {@link GetChangesetGroupListParams}.
+   * @returns {EntityListIterator<ChangesetGroup>} iterator for Changeset Group list, which internally queries entities in pages.
+   * See {@link EntityListIterator}, {@link ChangesetGroup}.
+   */
+  public getList(params: GetChangesetGroupListParams): EntityListIterator<ChangesetGroup> {
+    const entityCollectionAccessor = (response: unknown) => {
+      const changesetGroups = (response as ChangesetGroupsResponse).changesetGroups;
+      const mappedChangesetGroups = changesetGroups.map((changesetGroup) =>
+        this.appendRelatedEntityCallbacks(params.authorization, changesetGroup, params.headers));
+      return mappedChangesetGroups;
+    };
+
+    return new EntityListIteratorImpl(async () => this.getEntityCollectionPage<ChangesetGroup>({
+      authorization: params.authorization,
+      url: this._options.urlFormatter.getChangesetGroupListUrl({ iModelId: params.iModelId, urlParams: params.urlParams }),
+      entityCollectionAccessor,
+      headers: params.headers
+    }));
+  }
+
+  /**
+   * Gets a single Changeset Group identified by id. Wraps the
+   * {@link https://developer.bentley.com/apis/imodels-v2/operations/get-imodel-changeset-group-details/ Get iModel Changeset Group}
+   * operation from iModels API.
+   * @param {GetSingleChangesetGroupParams} params parameters for this operation. See {@link GetSingleChangesetGroupParams}.
+   * @returns {Promise<ChangesetGroup>} a Changeset Group with the specified id. See {@link ChangesetGroup}.
+   */
+  public async getSingle(params: GetSingleChangesetGroupParams): Promise<ChangesetGroup> {
+    const { authorization, iModelId, headers, changesetGroupId } = params;
```

Comment:
nitpick: this is a bit inconsistent. `getSingle` method implementations for various entities are almost identical, this follows the structure of `CheckpointOperations.getSingle` which is also a bit inconsistent, majority of implementations do not deconstruct `params`. would be good to unify this

Reply:
I see, seems like others (`Changeset` and `Checkpoints`) deconstruct the parameters to get one of the properties, which in this case does not apply.